File: docs/content/fundamentals/loops/code/loops.v1.full.py
"""Loops demo — read n numbers and print their sum.

Illustrates a ``for`` loop with the accumulator pattern: a variable declared
outside the loop (``total``) is updated on every iteration. This is the building
block of almost every program: repeat some work a fixed number of times.

Input:
    An integer ``n``, then ``n`` integers (whitespace-separated).
Output:
    A single line with the sum of the ``n`` integers.
"""
import sys


def main() -> None:
    # Read every whitespace-separated token at once: data[0] is n, then n values.
    data: list[str] = sys.stdin.read().split()
    n: int = int(data[0])

    total: int = 0                  # accumulator for the running sum
    for i in range(1, n + 1):       # `for`: iterate over the n values (data[1..n])
        total += int(data[i])       # convert token to int and add it

    print(total)


main()

File: docs/content/fundamentals/loops/code/loops.v1.full.cpp
/**
 * Loops demo — read n numbers and print their sum.
 *
 * Illustrates a `for` loop with the accumulator pattern: a variable declared
 * outside the loop (`sum`) is updated on every iteration. This is the building
 * block of almost every program: repeat some work a fixed number of times.
 *
 * Input:  an integer n, then n integers.
 * Output: a single line with the sum of the n integers.
 */
#include <iostream>
using namespace std;

int main() {
    int n;
    cin >> n;                       // how many numbers follow

    long long sum = 0;              // accumulator (long long avoids overflow)
    for (int i = 0; i < n; i++) {   // `for`: exactly n iterations, i = 0..n-1
        int x;
        cin >> x;                   // read the i-th number
        sum += x;                   // add it to the running total
    }

    cout << sum << "\n";
    return 0;
}
